azurerm/internal/services/datalake/data_lake_store_virtual_network_rule_resource.go

```diff
@@ -0,0 +1,154 @@
+package datalake
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/datalake/store/mgmt/2016-11-01/account"
+	"github.com/hashicorp/go-azure-helpers/response"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/datalake/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/datalake/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceDataLakeStoreVirtualNetworkRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDataLakeStoreVirtualNetworkRuleCreateUpdate,
+		Read:   resourceDataLakeStoreVirtualNetworkRuleRead,
+		Update: resourceDataLakeStoreVirtualNetworkRuleCreateUpdate,
+		Delete: resourceDataLakeStoreVirtualNetworkRuleDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.ValidateDataLakeStoreVirtualNetworkRuleName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"account_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.AccountName(),
+			},
+
+			"subnet_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceDataLakeStoreVirtualNetworkRuleCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Datalake.VirtualNetworkRulesClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	accountName := d.Get("account_name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	virtualNetworkSubnetId := d.Get("subnet_id").(string)
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	id := parse.NewVirtualNetworkRuleID(subscriptionId, resourceGroup, accountName, name)
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, accountName, name)
```

Comment:
could we instead use the id values here and throughout the rest of the function?
```suggestion
		existing, err := client.Get(ctx, id.ResourceGroup, id.AccountName, id.Name)
```